### ISSUE TITLE
Trailing whitespace produces bad markdown 

### ISSUE THREAD

**Issue by chooban** [inokawa/remark-slate-transformer#44]
I'll preface this by saying it's a case of bad input leading to bad output, so the issue might not be within this library.

I've created a repl here to show the problem: https://repl.it/@chooban/HoneydewHeartfeltQuadrant

Slate has allowed us to highlight and apply `strong` and `emphasis` style to trailing spaces on a word which is not so great. However, it renders fine in slate at the time of editing so it must be compensating. However, when we generate the markdown we get `is a **Japanese **manga artist` which is _technically_ correct, and when reading it back into a slate instance it ends up rendering like that without the strong tags. 

You can also recreate the issue on the storybook playground for this library by just adding a space into the line of nested formatting for bold, italic, etc.

Would you say that this issue is something that should be resolved within the transformer, or another part of the chain?

**Comment by inokawa** [comment]
Hi, @chooban 
I guess this issue should be solved by this lib.

It seems that `**Japanese **` is not a correct syntax in commonmark spec (try it in [this playground](https://commonmark.org/help/tutorial/02-emphasis.html)).
The commonmark spec is followed by [micromark](https://github.com/micromark/micromark), which is markdown parser used in remark.

However, `{ text: 'Japanese ', strong: true }` is correct syntax in slate.
So I think this lib should output correct markdown like `**Japanese** `.
I'll try to fix it soon.

**Comment by chooban** [comment]
That's great, thanks! I ended up implenting a fix in the code for now, so if it's of any use to you, here's my test case of Slate nodes to be serialized to markdown:

```
onst slate = [
      {
        type:     'paragraph',
        children: [
          {
            text: 'Naoki Urasawa is a ',
          },
          {
            text:   'Japanese ',
            strong: true,
          },
          {
            text: 'manga artist and musician. He has been ',
          },
          {
            emphasis: true,
            text:     'drawing ',
          },
          {
            text: 'manga since he ',
          },
          {
            strong: true,
            text:   ' was',
          },
          {
            text: ' four years old, and for most of his career has created two series simultaneously. ',
          },
          {
            strong: true,
            text:   'So there ',
          },
        ],
      },
    ]
```

**Comment by inokawa** [comment]
@chooban 
I fixed in 0.3.1. Thank you for your report.
